### PR TITLE
FE-1183 - Subaccount name shrink

### PR DESCRIPTION
**Pull request by nstanard**
### What Changed
 - `<Subaccount />` component has the shrinkLength prop that allows each spot to control the target cut-off length. 
 - Created a simple wrapper (currently being used in 2 places) to trim the subaccount name based on container width. 

### How To Test
 - Load up the app and look at all 8 spots that use the `<Subaccount />` making sure nothing is wrong and it looks good. 

### To Do
- [x] Adjust based on any feedback from the team

### CODE REVIEW

File: src/pages/domains/components/TrackingDomainsTable.js [hunk 1]
```diff
@@ -45,8 +45,13 @@ function MainCell({ row }) {
...
       {subaccountId && (
         <div>
-          <TranslatableText>Assignment: </TranslatableText>
-          <Subaccount all={sharedWithSubaccounts} id={subaccountId} name={subaccountName} />
+          <SubaccountShrinkWrapper
+            sharedWithSubaccounts={sharedWithSubaccounts}
+            subaccountId={subaccountId}
+            subaccountName={subaccountName}
+          >
+            <TranslatableText>Assignment: </TranslatableText>
```
Comment (shrirupa) [inline src/pages/domains/components/TrackingDomainsTable.js:53]: Font size for Assignment looks larger that before, we should have it as the same size as that "Added" below it
Screen Shot 2020-10-15 at 4 14 58 PM

Comment (shrirupa) [reply src/pages/domains/components/TrackingDomainsTable.js:53]: and TranslatableText should be moved out of `SubaccountWrapper`

File: src/pages/domains/components/SendingDomainsTable.js [hunk 2]
```diff
@@ -55,19 +55,24 @@ function MainCell({ row }) {
...
-        <TranslatableText>Assignment: </TranslatableText>
-
-        {subaccountId ? (
-          <Subaccount all={sharedWithSubaccounts} id={subaccountId} name={subaccountName} />
-        ) : (
-          'Primary Account'
-        )}
-      </Text>
+      {subaccountId ? (
+        <SubaccountShrinkWrapper
```
Comment (shrirupa) [inline src/pages/domains/components/SendingDomainsTable.js:59]: Font size for Assignment looks larger that before, we should have it as the same size as that "Added" below it
Screen Shot 2020-10-15 at 4 14 58 PM

Comment (shrirupa) [reply src/pages/domains/components/SendingDomainsTable.js:59]: and TranslatableText should be moved out of `SubaccountWrapper`

File: src/pages/domains/components/TrackingDomainsTable.js (hunk 1 above)
Comment (shrirupa) [inline src/pages/domains/components/TrackingDomainsTable.js:53]: we should probably move the `TranslatableText` outside SubaccountShrikWrapper

**Review by shrirupa — COMMENTED** [review]
Great work! just had few minor comments.

File: src/components/labels/Subaccount.js [hunk 3]
```diff
@@ -2,12 +2,16 @@ import React from 'react';
...
   let content = null;
   let defaultContent = null;
 
+  if (name && shrinkLength) {
+    name = shrinkToFit(name, shrinkLength);
+  }
+
   if (id) {
-    content = name ? `${shrinkToFit(name,12)} (${id})` : `Subaccount ${id}`;
+    content = name ? `${name} (${id})` : shrinkToFit(`Subaccount ${id}`, shrinkLength);
```
Comment (nstanard) [inline src/components/labels/Subaccount.js:14]: When there is an id but no name

image

File: src/pages/domains/components/SendingDomainsTable.js [hunk 4]
```diff
@@ -55,19 +55,24 @@ function MainCell({ row }) {
...
+            sharedWithSubaccounts={sharedWithSubaccounts}
+            subaccountId={subaccountId}
+            subaccountName={subaccountName}
+          >
+            <TranslatableText>Assignment: </TranslatableText>
+          </SubaccountShrinkWrapper>
+        </Text>
+      ) : (
+        <Text fontSize="200" lineHeight="200">
+          <TranslatableText>Assignment: Primary Account</TranslatableText>
```
Comment (nstanard) [inline src/pages/domains/components/SendingDomainsTable.js:70]: We could shrink this too right now it's not. 

image

File: src/components/labels/Subaccount.js (hunk 3 above)
Comment (nstanard) [reply src/components/labels/Subaccount.js:14]: When we have a name and id:
image

File: src/pages/domains/components/SendingDomainsTable.js [hunk 5]
```diff
@@ -55,19 +55,24 @@ function MainCell({ row }) {
...
-        {subaccountId ? (
-          <Subaccount all={sharedWithSubaccounts} id={subaccountId} name={subaccountName} />
-        ) : (
-          'Primary Account'
-        )}
-      </Text>
+      {subaccountId ? (
+        <Text fontSize="200" lineHeight="200">
+          <SubaccountShrinkWrapper
+            sharedWithSubaccounts={sharedWithSubaccounts}
```
Comment (shrirupa) [inline src/pages/domains/components/SendingDomainsTable.js:61]: Screen Shot 2020-10-19 at 1 20 40 PM
the subaccount in prod is cutoff

Comment (shrirupa) [reply src/pages/domains/components/SendingDomainsTable.js:61]: I think shrinkLength needs to be added to src/pages/sendingDomains/ListPage.js

Comment (shrirupa) [reply src/pages/domains/components/SendingDomainsTable.js:61]: same thing happening for /account/users
Screen Shot 2020-10-19 at 1 27 09 PM

Comment (shrirupa) [reply src/pages/domains/components/SendingDomainsTable.js:61]: Above is for src/pages/users/ListPage.js

Comment (shrirupa) [reply src/pages/domains/components/SendingDomainsTable.js:61]: same thing for /account/api-keys

**Review by shrirupa — APPROVED** [review]
Looks good to me!

**Pull request merged**